### PR TITLE
Run page button workflow test

### PR DESCRIPTION
**Pull request by ckendrick**
Adds a workflow test for the catalog, reduction, and all post-processing buttons on the run page.

This also passes through the django settings to docker compose, and switches system tests to use the `envtest` settings module.

A check was added to make sure the user has instrument permissions before sending processing requests.

Refer to [task INF177](https://code.ornl.gov/sns-hfir-scse/infrastructure/web-monitor/-/issues/177) and the instrument scientist [view use case](https://code.ornl.gov/sns-hfir-scse/infrastructure/web-monitor/-/wikis/Instrument-scientist-view).

### CODE REVIEW

File: tests/test_workflow.py [hunk 1]
```diff
@@ -88,3 +89,138 @@ def test(self):
...
+
+
+class TestPostProcessingWorkflow:
+    user = "InstrumentScientist"
+    pwd = "InstrumentScientist"
+    conn = None
+
+    def setup_class(cls):
+        cls.conn = psycopg2.connect(
+            database="workflow",
```
Comment (peterfpeterson) [inline tests/test_workflow.py:101]: Can we get this from the settings module similar to `tests/test_RunPageView.py`?
```python
from django.conf import settings

... 
cls.conn = psycopg2.connect(
database=settings.DATABASES["default"]["NAME"],
user=settings.DATABASES["default"]["USER"],
password=settings.DATABASES["default"]["PASSWORD"],
port=settings.DATABASES["default"]["PORT"],
host=settings.DATABASES["default"]["HOST"],
```
which is defined in `src/webmon_app/reporting/reporting_app/settings/base.py`. The only possible is if the app isn't "setup" so the settings aren't properly configured.

Maybe I'm being picky on this, but it makes it easier to synchronize things.

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
I would like to have checks for runs that don't exist as well to verify that errors are being handled correctly. I'm not sure if that is something that should be handled in the workflow tests or in the UI tests.

File: tests/test_workflow.py (hunk 1 above)
Comment (walshmm) [reply tests/test_workflow.py:101]: Currently the system test job ensures that the `unittest` settings module is active [here](https://github.com/neutrons/data_workflow/blob/next/.github/workflows/systemtests.yml#L27). 

Using the unittest module for running system tests is also documented as part of our developer procedure. 

So referring to a settings file for these bits of data should be pretty safe.

Comment (peterfpeterson) [reply tests/test_workflow.py:101]: If we change the code the documentation should be updated too.

**Review by peterfpeterson — APPROVED** [review]

File: tests/test_PostProcessWorkflow.py [hunk 2]
```diff
@@ -73,6 +74,7 @@ def get_status_errors(self, cursor, datarun_id):
                 errors.append(result)
         return errors
 
+    @unittest.skip("Skipping due to catalog start message not sending in CI")
```
Comment (peterfpeterson) [inline tests/test_PostProcessWorkflow.py:77]: Can you change this to [pytest](https://docs.pytest.org/en/6.2.x/skipping.html#skipping-test-functions)?

**Review by peterfpeterson — CHANGES_REQUESTED** [review]

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**